src/frontends/jax/src/op/convolution.cpp

```diff
@@ -0,0 +1,144 @@
+                                std::to_string(dimension_numbers.size()));
+    auto lhs_spec = dimension_numbers[0];
+    auto rhs_spec = dimension_numbers[1];
+    auto out_spec = dimension_numbers[2];
+    JAX_OP_CONVERSION_CHECK(lhs_spec.size() == rhs_spec.size() && lhs_spec.size() == out_spec.size(),
+                            "[JAX Frontend] internal error: specs in dimension_numbers must have the same size, but "
+                            "got lhs_spec.size() = " +
+                                std::to_string(lhs_spec.size()) +
+                                ", rhs_spec.size() = " + std::to_string(rhs_spec.size()) +
+                                ", out_spec.size() = " + std::to_string(out_spec.size()));
+    JAX_OP_CONVERSION_CHECK(lhs_spec.size() == 4,
+                            "[JAX Frontend] internal error: specs in dimension_numbers must have 4 elements.");
+
+    JAX_OP_CONVERSION_CHECK(spatial_dim == 2 || spatial_dim == 3,
+                            "[JAX Frontend] internal error: only 2D and 3D convolutions are supported.");
+    JAX_OP_CONVERSION_CHECK(padding.size() == 2, "[JAX Frontend] inconsistent model: padding must have 2 vectors.");
+    JAX_OP_CONVERSION_CHECK(padding[0].size() == spatial_dim,
+                            "[JAX Frontend] inconsistent model: padding vector must contain elements equal to "
+                            "doubled spatial dimensions ");
+    JAX_OP_CONVERSION_CHECK(input_dilation.size() == spatial_dim,
+                            "[JAX Frontend] inconsistent model: input_dilation vector must contain elements equal to "
+                            "spatial dimensions ");
+    JAX_OP_CONVERSION_CHECK(kernel_dilation.size() == spatial_dim,
+                            "[JAX Frontend] inconsistent model: kernel_dilation vector must contain elements equal to "
+                            "spatial dimensions ");
+    JAX_OP_CONVERSION_CHECK(batch_group_count == 1,
+                            "[JAX Frontend] Convolution is supported only with batch_group_count equal to one yet.");
+
+    // Tranpose input and kernel to NCHW format
+    if (need_transpose(lhs_spec)) {
+        auto input_transpose_order = std::make_shared<v0::Constant>(element::i64, Shape{lhs_spec.size()}, lhs_spec);
+        input = std::make_shared<v1::Transpose>(input, input_transpose_order);
+    }
+    if (need_transpose(rhs_spec)) {
+        auto kernel_transpose_order = std::make_shared<v0::Constant>(element::i64, Shape{rhs_spec.size()}, rhs_spec);
+        kernel = std::make_shared<v1::Transpose>(kernel, kernel_transpose_order);
+    }
+
+    Strides strides(spatial_dim);
+    Strides dilations(spatial_dim);
+    CoordinateDiff pads_begin(spatial_dim);
+    CoordinateDiff pads_end(spatial_dim);
+    for (size_t ind = 0; ind < spatial_dim; ++ind) {
+        strides[ind] = static_cast<size_t>(window_strides[ind]);
```

Comment:
What if `window_strides` is less then `spatial_dim`? Also I would suggest to do conversion in the `const_named_param` method (or `get_attribute` if you will use it as I suggested previously) by specifying template for `Strides`

Reply:
The `spatial_dim` is exactly got from the size of `window_strides` so they are always the same. :)